### PR TITLE
feat: create data for build definition

### PR DESCRIPTION
**Pull request by josh-barker**
## All Submissions:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.
* [x] My code follows the code style of this project.
* [x] I ran lint checks locally prior to submission.
* [x] Have you checked to ensure there aren't other open PRs for the same update/change?

## What about the current behavior has changed?

Create data for build definition.

Note: this returns a **single** build definition, not multiples.

Issue Number: #562 

## Does this introduce a change to `go.mod`, `go.sum` or `vendor/`?

- [ ] Yes
- [x] No

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: website/docs/d/build_definition.html.markdown [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+* `exclude` - A `exclude` block as defined below.
+
+* `include` - A `include` block as defined below.
+
+---
+
+A `ci_trigger` block exports the following:
+
+* `override` - A `override` block as defined below.
```
Comment (xuzhang3) [inline website/docs/d/build_definition.html.markdown:75]: `ci_trigger.override` is different from `pull_request_trigger.override`

File: website/docs/d/build_definition.html.markdown [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+* `yml_path` - The path of the Yaml file describing the build definition.
+
+---
+
+A `schedules` block exports the following:
+
+* `branch_filter` - A `branch_filter` block as defined above.
+
+* `days_to_build` - A `days_to_build` block as defined above.
```
Comment (xuzhang3) [inline website/docs/d/build_definition.html.markdown]: `days_to_build` block not defined

**Review by xuzhang3 — CHANGES_REQUESTED** [review]
@josh-barker  a few changes requested, others LGTM.
`website/docs/d/build_definition.html.markdown` should add to `azuredevops.erb`

**Comment by xuzhang3** [comment]
```log
=== RUN   TestAccBuildDefinition_DataSource
=== PAUSE TestAccBuildDefinition_DataSource
=== CONT  TestAccBuildDefinition_DataSource
--- PASS: TestAccBuildDefinition_DataSource (89.11s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        89.926s
=== RUN   TestAccBuildDefinition_with_path_DataSource
=== PAUSE TestAccBuildDefinition_with_path_DataSource
=== CONT  TestAccBuildDefinition_with_path_DataSource
--- PASS: TestAccBuildDefinition_with_path_DataSource (38.27s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        38.721s

```

**Comment by josh-barker** [comment]
Hi @xuzhang3, I've address your feedback.

Cheers

**Comment by xuzhang3** [comment]
@josh-barke  I run acc test and got error below,  `computed=true`  is for data source not for resource. In resource, `branch_filter` etc. is configurable. 
```log
 resource_build_definition_test.go:162: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: Value for unconfigurable attribute
        
          with azuredevops_build_definition.build,
          on terraform_plugin_test.tf line 26, in resource "azuredevops_build_definition" "build":
          26:       branch_filter {
        
        Can't configure a value for "ci_trigger.0.override.0.branch_filter": its
        value will be decided automatically based on the result of applying this
        configuration.
        
        Error: Value for unconfigurable attribute
        
          with azuredevops_build_definition.build,
          on terraform_plugin_test.tf line 29, in resource "azuredevops_build_definition" "build":
          29:       path_filter {
        
        Can't configure a value for "ci_trigger.0.override.0.path_filter": its value
        will be decided automatically based on the result of applying this
        configuration.
        
        Error: Value for unconfigurable attribute
        
          with azuredevops_build_definition.build,
          on terraform_plugin_test.tf line 38, in resource "azuredevops_build_definition" "build":
          38:     branch_filter {
        
        Can't configure a value for "schedules.0.branch_filter": its value will be
        decided automatically based on the result of applying this configuration.

```

**Comment by josh-barker** [comment]
Hey @xuzhang3, that's been resolved now.
I also fixed `TestAccBuildDefinition_Create_Update_Import`. The repository reference was set to `master`, but AZDO returned `refs/heads/master`, which caused an unstable plan.

Cheers

**Comment by xuzhang3** [comment]
@josh-barker LGTM and I updated some of the doc formatting. 
```log
=== RUN   TestAccBuildDefinition_DataSource
=== PAUSE TestAccBuildDefinition_DataSource
=== RUN   TestAccBuildDefinition_with_path_DataSource
=== PAUSE TestAccBuildDefinition_with_path_DataSource
=== RUN   TestAccBuildDefinition_Create_Update_Import
=== PAUSE TestAccBuildDefinition_Create_Update_Import
=== RUN   TestAccBuildDefinition_WithVariables_CreateAndUpdate
=== PAUSE TestAccBuildDefinition_WithVariables_CreateAndUpdate
=== RUN   TestAccBuildDefinition_Schedules
=== PAUSE TestAccBuildDefinition_Schedules
=== CONT  TestAccBuildDefinition_DataSource
=== CONT  TestAccBuildDefinition_WithVariables_CreateAndUpdate
=== CONT  TestAccBuildDefinition_Schedules
=== CONT  TestAccBuildDefinition_Create_Update_Import
=== CONT  TestAccBuildDefinition_with_path_DataSource
--- PASS: TestAccBuildDefinition_with_path_DataSource (92.81s)
--- PASS: TestAccBuildDefinition_Schedules (94.19s)
--- PASS: TestAccBuildDefinition_WithVariables_CreateAndUpdate (131.96s)
--- PASS: TestAccBuildDefinition_Create_Update_Import (140.70s)
--- PASS: TestAccBuildDefinition_DataSource (148.15s)
PASS
ok      github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/acceptancetests        148.986s

```

**Review by xuzhang3 — APPROVED** [review]

**Comment by josh-barker** [comment]
Thanks @xuzhang3 ! 🙂

**Comment by josh-barker** [comment]
Hey @xuzhang3, just wondering when the next release will be?

**Comment by xuzhang3** [comment]
@josh-barker  will coming soon, working on it.

**Pull request merged**